python/update_package_properties.py

```diff
@@ -0,0 +1,311 @@
+import argparse
+import re
+from datetime import date, datetime
+
+import pathlib2
+import string_utils
+from github import Github
+from jinja2 import Environment, FileSystemLoader
+from parameters_validation import (no_whitespaces, non_blank, non_empty, non_negative, validate_parameters,
+                                   parameter_validation)
+from dataclasses import dataclass
+
+BASE_PATH = pathlib2.Path(__file__).parent.absolute()
+
+project_name_suffix_dict = {"citus": "citus", "citus-enterprise": "citus",
+                            "pg-auto-failover": "", "pg-auto-failover-enterprise": "", "pg-cron": "", "pg-xn": ""}
+
+
+@parameter_validation
+def is_version(version: str):
+    if version is None or not version:
+        raise ValueError("version should be non-empty and should not be None")
+    if not re.match(r"\d+\.\d+\.\d+$", version):
+        raise ValueError(
+            "version should include 3 levels of versions consists of numbers  separated with dots. e.g: 10.0.1")
+
+
+@parameter_validation
+def is_tag(tag: str):
+    if tag is None or not tag:
+        raise ValueError("tag  should be non-empty and should not be None")
+    if not re.match(r"v\d+\.\d+\.\d+$", tag):
+        raise ValueError(
+            "tag  should start with v and should include 3 levels of versions consists of numbers " +
+            "separated with dots. e.g: v10.0.1")
+
+
+@parameter_validation
+def is_email(email: str):
+    if not string_utils.is_email(email):
+        raise ValueError("Parameter is not in email format")
+
+
+@parameter_validation
+def is_project_changelog_header(header: str):
+    if header is None or not header:
+        raise ValueError("header should be non-empty and should not be None")
+    if not re.match(r"^### \w+\sv\d+\.\d+\.\d+\s\(\w+\s\d+,\s\d+\)\s###$", header):
+        raise ValueError(
+            f"changelog header is in invalid format. Actual:{header} Expected: ### citus v8.3.3 (March 23, 2021) ### ")
+
+
+@dataclass
+class PackagePropertiesParams:
+    project_name: str = ""
+    project_version: str = ""
+    fancy: bool = False
+    fancy_version_number: int = 0
+    microsoft_email: str = ""
+    name_surname: str = ""
+    changelog_date: datetime = datetime.now()
+    latest_changelog: str = ""
+
+    def __init__(self, project_name: str, project_version: str, fancy: bool, fancy_version_number: int,
+                 microsoft_email: str = "", name_surname: str = "", changelog_date: datetime = datetime.now(),
+                 latest_changelog: str = ""):
+        self.project_name = project_name
+        self.project_version = project_version
+        self.fancy = fancy
+        self.fancy_version_number = fancy_version_number
+
+        self.microsoft_email = microsoft_email
+        self.name_surname = name_surname
+        self.changelog_date = changelog_date
+        self.latest_changelog = latest_changelog
```

Comment:
No need to add a `__init__()` for a dataclass.
```suggestion
```

Reply:
Fixed